src/mpf.rs

```diff
@@ -119,13 +120,26 @@ impl Mpf {
         }
     }
 
-    pub fn get_str(&mut self, n_digits: i32, base: i32, exp: &mut c_long) -> String{
-        let c_str = CString::new("").unwrap();
-        let out;
-        unsafe{
-            out = CString::from_raw(__gmpf_get_str(c_str.into_raw(), exp, base, n_digits, &mut self.mpf));
+    pub fn get_str(&self, n_digits: i32, base: i32, exp: &mut c_long) -> String {
+        use std::ptr::null_mut;
+        if n_digits == 0 {
+            // Maximal significant digits requested. Let GMP compute the length
+            // and allocate the space required.
+            unsafe {
+                let p = __gmpf_get_str(null_mut(), exp, base, n_digits, &self.mpf);
+                // De-allocates the GMP string on drop.
+                GString::from_raw(p).to_str().unwrap().to_string()
+            }
+        } else {
+            let bytes = unsafe {
+                // n_digits + 2 from mpf/get_str.c.
+                let mut bytes: Vec<u8> = vec![uninitialized(); n_digits as usize + 2];
+                let c_str: *mut c_char = bytes.as_mut_ptr() as *mut c_char;
+                __gmpf_get_str(c_str, exp, base, n_digits, &self.mpf);
+                bytes
+            };
+            String::from_utf8(bytes).unwrap()
```

Comment:
yeah, i realized what you meant a few minutes after hitting send. this branch should be tested anyway, and it shows up in the test.